### PR TITLE
docker: switch to multistage build

### PR DESCRIPTION
**Pull request by mccutchen**
This results in a minimal final image without the full golang runtime, addressing #20.

RFR @loganmeetsworld @shrayolacrayon 
cc @danbf

### CODE REVIEW

File: Godeps [hunk 1]
```diff
@@ -1,7 +1,6 @@
 github.com/18F/hmacauth                  1.0.1
 gopkg.in/yaml.v2                         v2
 github.com/imdario/mergo                 v0.3.4
-github.com/BurntSushi/toml               d94612f9fc140360834f9742158c70b5c5b5535b
```
Comment (loganmeetsworld) [inline Godeps]: Thought we'd already removed this!

**Comment by mccutchen** [comment]
Size difference, for posterity:

```
$ docker image list | grep sso
sso                             master              57eb2a5f0f46        About a minute ago       1.13GB
sso                             alpine              cc20033f631e        10 minutes ago           26.3MB
```

Went from 1.13GB to 26.3MB!

**Comment by mccutchen** [comment]
Check latest commit, we need to add the `ca-certificates` package to the final image.  Without it, we get an error trying to talk to Google:

```
{"error":"Post https://www.googleapis.com/oauth2/v3/token: x509: failed to load system roots and no roots provided","level":"error","msg":"error redeeming authentication code","remote_address":"172.20.0.1","service":"sso-auth","time":"2018-08-23 00:57:22.82312"}
```

**Comment by mccutchen** [comment]
Another update:  Apparently something about the alpine image prevents the pure Go DNS resolver from properly reading `/etc/hosts`, which breaks the quick start:
https://groups.google.com/d/msg/golang-nuts/G-faJ0bthz0/Lx-MsDIOCAAJ

Now we've updated to `debian:stable-slim` instead of `alpine:3.8` for the base image, bringing our final image size up to 83MB:

```
sso                             debian              888f18f0bcc5        39 seconds ago      83.5MB
```

**Comment by danbf** [comment]
wow, great size savings!

**Comment by mccutchen** [comment]
Yep! Just wanted to makes sure y’all had a chance to see the individual
fixes before I squashed them away!

**Pull request merged**